Content.Client/Info/PlaytimeStats/PlaytimeStatsWindow.cs

```diff
@@ -0,0 +1,185 @@
+
+        foreach (var entry in sortedEntries)
+        {
+            var styleBox = new StyleBoxFlat { BackgroundColor = _useAltColor ? _altColor : _defaultColor };
+            entry.UpdateShading(styleBox);
+            RolesPlaytimeList.AddChild(entry);
+            _useAltColor ^= true;
+        }
+    }
+
+    private void SortByPlaytime(PlaytimeStatsHeader.SortDirection direction)
+    {
+        var header = RolesPlaytimeList.GetChild(0) as PlaytimeStatsHeader;
+
+        var entries = RolesPlaytimeList.Children.OfType<PlaytimeStatsEntry>().ToList();
+
+        RolesPlaytimeList.RemoveAllChildren();
+
+        if (header != null)
+            RolesPlaytimeList.AddChild(header);
+
+        var sortedEntries = (direction == PlaytimeStatsHeader.SortDirection.Ascending)
+            ? entries.OrderBy(entry => entry.PlaytimeText != null ? ParsePlaytime(entry.PlaytimeText) : TimeSpan.Zero)
+                .ToList()
+            : entries.OrderByDescending(entry =>
+                entry.PlaytimeText != null ? ParsePlaytime(entry.PlaytimeText) : TimeSpan.Zero).ToList();
+
+        _useAltColor = false;
+
+        foreach (var entry in sortedEntries)
+        {
+            var styleBox = new StyleBoxFlat { BackgroundColor = _useAltColor ? _altColor : _defaultColor };
+            entry.UpdateShading(styleBox);
+            RolesPlaytimeList.AddChild(entry);
+            _useAltColor ^= true;
+        }
+    }
+
+
+    private TimeSpan ParsePlaytime(string playtimeString)
+    {
+        var localisedFilterParams = Loc.GetString("ui-playtime-time-format");
+
+        // Replace placeholders with regex capture groups
+        var regexPattern = localisedFilterParams
+            .Replace("{$hours}", @"(?<hours>\d+)")
+            .Replace("{$minutes}", @"(?<minutes>\d+)");
+
+        var regex = new Regex(regexPattern);
+        var match = regex.Match(playtimeString);
+
+        if (match.Success &&
+            int.TryParse(match.Groups["hours"].Value, out var hours) &&
+            int.TryParse(match.Groups["minutes"].Value, out var minutes))
+        {
+            return new TimeSpan(hours, minutes, 0);
+        }
+
+        _sawmill.Error($"Couldn't ParsePlaytime string: {playtimeString} Reverting to zero time for entry");
+        return TimeSpan.Zero;
+    }
+
+
+
+    private void PopulatePlaytimeData()
+    {
+        try
+        {
+            var overallPlaytime = _jobRequirementsManager.FetchOverallPlaytime();
+
+            if (overallPlaytime is var timeSpan)
+            {
+                var formattedPlaytime = ConvertTimeSpanToHoursMinutes(timeSpan);
+                OverallPlaytimeLabel.Text = Loc.GetString("ui-playtime-overall", ("time", formattedPlaytime));
+            }
+
+            var rolePlaytimes = _jobRequirementsManager.FetchPlaytimeByRoles();
+
+            RolesPlaytimeList.RemoveAllChildren();
+            PopulatePlaytimeHeader();
+
+            foreach (var rolePlaytime in rolePlaytimes)
+            {
```

Comment:
You could just make it IEnumerable and avoid allocating the entire dictionary instead.